docs/advanced_usage.rst

```diff
@@ -34,6 +34,20 @@ hooks, as these can be run on any platform. However, if you intend for your
 template to only be run on a single platform, a shell script (or `.bat` file
 on Windows) can be a quicker alternative.
 
+By default, hooks are duplicated in a temporary file and the template is directly applied in this ephemeral file. This allows you to use template variables directly within your hook, as show in `Example: Validating template variables`_ below.
+
+Besides this, the cookiecutter context object is serialized and passed to your hook through the standard input stream. You may then want to modify some configuration settings and send them back to the main context by writing to the standard output stream.
+
```

Comment:
I would suggest removing "Besides this" here.
And "You may then want to" => "You can then" ?